openmetadata-service/src/main/java/org/openmetadata/service/search/elasticsearch/ElasticSearchClient.java

```diff
@@ -757,6 +757,123 @@ public Response searchLineage(
     return Response.status(OK).entity(responseMap).build();
   }
 
+  private void getEntityRelationship(
+      String fqn,
+      int depth,
+      Set<Map<String, Object>> edges,
+      Set<Map<String, Object>> nodes,
+      String queryFilter,
+      String direction,
+      boolean deleted)
+      throws IOException {
+    if (depth <= 0) {
```

Comment:
how can be depth be below 0 here? we should throw an error not return

Reply:
This is a function that will be called recursively. The depth is there for handling that
E.g by default depth with start at 3 and then with each recursion it'll get decremented unless it reaches 0